app/src/main/java/org/wikipedia/dataclient/donate/CampaignCollection.kt

```diff
@@ -0,0 +1,50 @@
+package org.wikipedia.dataclient.donate
+
+import kotlinx.coroutines.Dispatchers
+import kotlinx.coroutines.withContext
+import kotlinx.serialization.Serializable
+import kotlinx.serialization.json.JsonElement
+import kotlinx.serialization.json.decodeFromJsonElement
+import okhttp3.Request
+import org.wikipedia.dataclient.okhttp.OkHttpConnectionFactory
+import org.wikipedia.json.JsonUtil
+import org.wikipedia.settings.Prefs
+import org.wikipedia.util.GeoUtil
+import java.time.LocalDateTime
+
+object CampaignCollection {
+
+    private const val CAMPAIGN_VERSION = 1
+
+    private const val CAMPAIGNS_URL = "https://donate.wikimedia.org/wiki/MediaWiki:AppsCampaignConfig.json?action=raw"
+    private const val CAMPAIGNS_URL_DEBUG = "https://donate.wikimedia.org/wiki/MediaWiki:AppsCampaignConfigStaging.json?action=raw"
+
+    suspend fun getActiveCampaigns(): List<Campaign> {
+        val campaignList = mutableListOf<Campaign>()
+
+        withContext(Dispatchers.IO) {
+            val request = Request.Builder().url(CAMPAIGNS_URL_DEBUG).build()
+            val response = OkHttpConnectionFactory.client.newCall(request).execute()
+            val json = response.body!!.string()
```

Comment:
Can we use `null-safey` here?

Reply:
yep. (and also updated to the newest actions-array structure)